pkg/issuer/acme/dns/dns.go

```diff
@@ -345,6 +345,8 @@ func (s *Solver) solverForChallenge(ctx context.Context, issuer v1.GenericIssuer
 			providerConfig.AzureDNS.HostedZoneName,
 			s.DNS01Nameservers,
 			canUseAmbientCredentials,
+			providerConfig.AzureDNS.ManagedIdentity.ClientID,
+			providerConfig.AzureDNS.ManagedIdentity.ResourceID,
```

Comment:
`providerConfig.AzureDNS.ManagedIdentity` could be `nil`.

Check for `nil` pointer  `providerConfig.AzureDNS.ManagedIdentity` or pass `providerConfig.AzureDNS.ManagedIdentity` pointer instead of the individual fields and check for nil inside the function.

Reply:
I'm not passing the struct pointer and checking for null inside, hope that fixes it.